### DISCUSSION TITLE
How to query one to many relation is there is more than 0 entries on the many side?

### DISCUSSION THREAD

**Discussion by Scalahansolo** [prisma/prisma#19676] [Q&A] [answered]
Lets say I have two models `Posts` and `Comments`. Where a `Post` has 0 or more `Comments`.

Does Prisma have a way to query for all posts that where the number of comments is equal to 0? I would also like to query for all Posts have have more than 0 comments.

**Comment by ludralph** [comment, marked as answer]
Hi @Scalahansolo 👋 

> Does Prisma have a way to query for all posts where the number of comments is equal to 0?

You can use `none` to return ` Post` with zero `Comments` as documented [here](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#filter-on-absence-of--to-many-records). For example:
```js
  await prisma.post.findMany({
  where: {
    comments: {
      none: {},
    },
  },
  include: {
    comments: true,
  },
})
```

> I would also like to query for all Posts have more than 0 comments.

You can use `some` to return `Post` with at least one `Comment` as documented [here](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#filter-on-presence-of-related-records). For example:
```js
await prisma.post.findMany({
  where: {
    comments: {
      some: {},
    },
  },
  include: {
   comments: true,
  },
})
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Scalahansolo** [reply]
Ahh this is perfect! I was under the assumption you had to pass in values for those. Thanks for the help!
